src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+            loadSamLaunchConfig(postMessageFn)
+            break
+        case 'getSamplePayload':
+            getSamplePayload(postMessageFn)
+            break
+        case 'getTemplate':
+            getTemplate(postMessageFn)
+            break
+        case 'saveLaunchConfig':
+            saveLaunchConfig(message.data.launchConfig)
+            break
+        case 'invokeLaunchConfig':
+            invokeLaunchConfig(message.data.launchConfig, context)
+            break
+    }
+}
+
+/**
+ * Open a quick pick containing the names of launch configs in the `launch.json` array.
+ * Filter out non-supported launch configs.
+ * Call back into the webview with the selected launch config.
+ * @param postMessageFn
+ */
+async function loadSamLaunchConfig(postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>): Promise<void> {
+    // TODO: Find a better way to infer this. Might need another arg from the frontend (depends on the context in which the launch config is made?)
+    const workspaceFolder = vscode.workspace.workspaceFolders?.length ? vscode.workspace.workspaceFolders[0] : undefined
+    if (!workspaceFolder) {
+        // TODO Localize
+        vscode.window.showErrorMessage(localize('AWS.lambda.form.noFolder', 'No workspace folder found.'))
+        return
+    }
+    const uri = workspaceFolder.uri
+    const launchConfig = new LaunchConfiguration(uri)
+    const pickerItems = getLaunchConfigQuickPickItems(launchConfig, uri)
+    if (pickerItems.length === 0) {
+        pickerItems.push({
+            index: -1,
+            label: localize('AWS.lambda.form.noDebugConfigs', 'No aws-sam debug configurations found.'),
+            detail: localize('AWS.picker.dynamic.noItemsFound.detail', 'Click here to go back'),
+        })
+    }
+    const qp = picker.createQuickPick({
+        items: pickerItems,
+        options: {
+            title: localize('AWS.lambda.form.selectDebugConfig', 'Select Debug Configuration'),
+        },
+    })
+
+    const choices = await picker.promptUser({
+        picker: qp,
+    })
+    const pickerResponse = picker.verifySinglePickerOutput<LaunchConfigPickItem>(choices)
+
+    if (!pickerResponse || pickerResponse.index === -1) {
+        return
+    }
+    postMessageFn({
+        command: 'loadSamLaunchConfig',
+        data: {
+            launchConfig: pickerResponse.config!,
+        },
+    })
+}
+
+interface SampleRequestManifest {
+    requests: {
+        request: {
+            name?: string
+            filename?: string
+        }[]
+    }
+}
+
+interface SampleQuickPickItem extends vscode.QuickPickItem {
+    filename: string
+}
+
+/**
+ * Open a quick pick containing upstream sample payloads.
+ * Call back into the webview with the contents of the payload to add to the JSON field.
+ * @param postMessageFn
+ */
+async function getSamplePayload(postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>): Promise<void> {
+    // stolen from invokeLambda.ts
```

Comment:
extract the duplicated code to a common method `resourcefetcher.ts` .

(Doesn't matter that `resourcefetcher.ts` is not deeply related to "lambda samples". It is easier to explode than to synthesize, so let's synthesize while we have the chance.)

Reply:
Yeah, this was hackathon leftovers. I'll do some proper cleaning now that it's merging into master.